pkg/clients/ec2/securitygroup.go

```diff
@@ -34,3 +39,58 @@ func IsSecurityGroupNotFoundErr(err error) bool {
 	}
 	return false
 }
+
+// IsRuleAlreadyExistsErr returns true if the error is because the rule already exists.
+func IsRuleAlreadyExistsErr(err error) bool {
```

Comment:
Is the `InvalidPermissionDuplicate` code used elsewhere? Should this be `IsInvalidPermissionDuplicateErr` in case we want to use it for other AWS errors with this code?

Reply:
It's not used elsewhere but I think having `AlreadyExists` in the name sounds more familiar because actual error changes from one resource to another.